### PR TITLE
Update new query and parser for zig

### PR DESCRIPTION
**Pull request by maxxnino**
# Info
I've been working on a new parser for zig since 8 days ago. It's base on zig-spec so the grammar and query really stable. I tested with [test cases](https://github.com/ziglang/zig-spec/tree/master/grammar/tests) in zig-spec. Passed 87/93. The last [6 of them](https://github.com/maxxnino/tree-sitter-zig/issues/1) with only one small [issue](https://github.com/maxxnino/tree-sitter-zig/issues/3). I've been using nvim-treesitter as a referent to make query for highlight, indent and fold. Small progress in local. I tested with my neovim, every working really good. 

# Progress
- [X] Highlight
- [x] Indent
- [x] Fold
- [ ] Local

---
# Highlight
demox

# Fold
foldsx

### CODE REVIEW

File: queries/zig/highlights.scm [hunk 1]
```diff
@@ -1,147 +1,159 @@
...
 ] @boolean
 
-"return" @keyword.return
+[
+  "undefined"
+  "unreachable"
+] @constant.builtin
 
-"fn" @keyword.function
+"null" @constant.macro
```
Comment (theHamsta) [inline queries/zig/highlights.scm:154]: Why macro?

File: queries/zig/highlights.scm [hunk 2]
```diff
@@ -152,71 +164,94 @@
...
-] @punctuation.bracket
+  "struct"
+  "enum"
+  "union"
+  "error"
+  "packed"
+  "opaque"
+] @keyword
+
+"error" @exception
```
Comment (theHamsta) [inline queries/zig/highlights.scm:182]: What about `try` being exception?

**Review by theHamsta — COMMENTED** [review]
Looks good to me!

@Himujjal any thoughts on that?

File: lua/nvim-treesitter/parsers.lua [hunk 3]
```diff
@@ -633,11 +633,11 @@ list.bibtex = {
 
 list.zig = {
   install_info = {
-    url = "https://github.com/Himujjal/tree-sitter-zig",
+    url = "https://github.com/maxxnino/tree-sitter-zig",
     files = { "src/parser.c" },
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:637]: You need to add `branch="main"`

File: queries/zig/highlights.scm (hunk 2 above)
Comment (maxxnino) [reply queries/zig/highlights.scm:182]: The hl for keyword i use this https://github.com/ziglang/zig.vim/blob/master/syntax/zig.vim as a reference. If this is make it better, i could make a change. Just wait me a bit

File: lua/nvim-treesitter/parsers.lua (hunk 3 above)
Comment (maxxnino) [reply lua/nvim-treesitter/parsers.lua:637]: sorry, i don't know where to add it

**Comment by Himujjal** [comment]
I have to go through the commit once.

 Last time I built it, it was terribly slow in performance and TS DLL size. A reimplementation had to be made anyways.

Anyways. I am not much active on this project and probably won't be next 2-3 months. If @maxxnino wants to take over. Good!

File: queries/zig/highlights.scm (hunk 2 above)
Comment (theHamsta) [reply queries/zig/highlights.scm:182]: I meant `try` could be `@keyword.exception` since it is a keyword related to exception/error handling.

**Review by theHamsta — APPROVED** [review]
Looks good to me. I think we can merge as soon as CI passes.

**Comment by maxxnino** [comment]
> Looks good to me. I think we can merge as soon as CI passes.

It failed with query `@annotation`. I thought that in the highlight group

File: lua/nvim-treesitter/parsers.lua (hunk 3 above)
Comment (stsewd) [reply lua/nvim-treesitter/parsers.lua:637]: you need to add it on the `install_info` table, you can search for `branh` on this same file for reference.

File: queries/zig/injections.scm [hunk 4]
```diff
@@ -1 +0,0 @@
-(line_comment) @comment
```
Comment (stsewd) [inline queries/zig/injections.scm:1]: I think you can update the injections with the new comment nodes.

**Comment by stsewd** [comment]
> It failed with query `@annotation`. I thought that in the highlight group

We have `@attribute`, maybe that's what you are looking for https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#keywords

**Comment by maxxnino** [comment]
Can you check the CI. I don't know what cause the problem

**Comment by stsewd** [comment]
> Can you check the CI. I don't know what cause the problem

Looks like you need to update the lockfile with the correct commit version of your parser https://github.com/nvim-treesitter/nvim-treesitter/blob/2b51d9b766c34128db22484c343709e19f5301b5/lockfile.json#L209-L211

**Comment by maxxnino** [comment]
> Looks like you need to update the lockfile with the correct commit version of your parser
> 
> https://github.com/nvim-treesitter/nvim-treesitter/blob/2b51d9b766c34128db22484c343709e19f5301b5/lockfile.json#L209-L211

Thank you, I updated the lockfile. Seem to passed in unix like system. But still failed on window

**Review by vigoux — APPROVED** [review]
I think that adding locals could wait for another time.

**Comment by maxxnino** [comment]
> I think that adding locals could wait for another time.

Yes, that would be nice. Because, i need to learn more about locals anyway

**Pull request merged**